mindsdb_native/libs/phases/data_transformer/data_transformer.py

```diff
@@ -1,88 +1,99 @@
-            try:
-                date = datetime.datetime.utcfromtimestamp(date_str)
-            except:
-                return None
-        return date.strftime('%Y-%m-%d')
-
-    @staticmethod
-    def _standardize_datetime(date_str):
-        try:
-            # will return a datetime object
-            dt = parse_datetime(str(date_str))
-        except:
-            try:
-                dt = datetime.datetime.utcfromtimestamp(date_str)
-            except:
-                return None
-
-        return dt.strftime('%Y-%m-%d %H:%M:%S')
-
-    @staticmethod
-    def _lightwood_datetime_processing(dt):
-        dt = pd.to_datetime(dt, errors = 'coerce')
+
+def _standardize_date(date_str):
+    try:
+        # will return a datetime object
+        date = parse_datetime(date_str)
+    except Exception:
         try:
-            return dt.timestamp()
-        except:
+            date = datetime.datetime.utcfromtimestamp(date_str)
+        except Exception:
             return None
+    return date.strftime('%Y-%m-%d')
+
 
-    @staticmethod
-    def clean_float_or_none(val):
+def _standardize_datetime(date_str):
+    try:
+        # will return a datetime object
+        dt = parse_datetime(str(date_str))
+    except Exception:
         try:
-            return clean_float(val)
-        except:
+            dt = datetime.datetime.utcfromtimestamp(date_str)
+        except Exception:
             return None
 
-    def _aply_to_all_data(self, input_data, column, func, transaction_type):
-        if transaction_type == TRANSACTION_LEARN:
-            input_data.train_df[column] = input_data.train_df[column].apply(func)
-            input_data.test_df[column] = input_data.test_df[column].apply(func)
-            input_data.validation_df[column] = input_data.validation_df[column].apply(func)
+    return dt.strftime('%Y-%m-%d %H:%M:%S')
+
+
+def _lightwood_datetime_processing(dt):
+    dt = pd.to_datetime(dt, errors='coerce')
+    try:
+        return dt.timestamp()
+    except Exception:
+        return None
 
 
+def _clean_float_or_none(val):
+    try:
+        return clean_float(val)
+    except Exception:
+        return None
+
+
+class DataTransformer(BaseModule):
+    def remove_duplicate_rows_from_all_data(self, input_data):
+        duplicated = input_data.data_frame.duplicated(keep='first')
+        duplicate_idx = input_data.data_frame.index[duplicated]
+
+        input_data.data_frame.drop(duplicate_idx, inplace=True)
+        if input_data.train_idx is not None:
+            input_data.train_idx = np.array([i for i in input_data.train_idx if i not in duplicate_idx])
+            input_data.validation_idx = np.array([i for i in input_data.validation_idx if i not in duplicate_idx])
+            input_data.test_idx = np.array([i for i in input_data.test_idx if i not in duplicate_idx])
+
+        no_dropped = sum(duplicated)
+        if no_dropped > 0:
+            self.log.warning(f'Dropped {no_dropped} duplicate rows.')
```

Comment:
In principle the logic here works, might even work if you switch back to having dataframes instead of a collection of indexes in the input_data.

Reply:
Hmmm, true, actually if `train_df` and other split dataframes preserve the initial index it can work